### PR TITLE
 chore: fixed cache on solutions related routes and removed redundant queries

### PR DESCRIPTION
**Pull request by zaCKoZAck0**
Fixed:
1. Cache invalidation / query re-fetching on Solution Add / Delete
2. Removed redundant queries doing the same thing 

## Related Issue

Closes #

## How Has This Been Tested?

1. Add a solution
Check for updates in solution row
2. Delete a solution
Check for updates in solution row

### CODE REVIEW

File: apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/delete/index.tsx [hunk 1]
```diff
@@ -36,6 +40,14 @@ export function SolutionDeleteDialog({ children, solution, ...props }: SolutionD
         variant: 'destructive',
         description: 'An error occurred while trying to delete the comment.',
       });
+    } finally {
+      // invalidate cache on deleting a solution successfully
+      queryClient.invalidateQueries({
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/delete/index.tsx:45]: we only want to invalidate on success i would think

File: apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-editor.tsx [hunk 2]
```diff
@@ -77,8 +77,13 @@ export function SolutionEditor({ dismiss, challenge }: Props) {
         title: 'Uh oh! Something went wrong. Please try again.',
       });
     } finally {
-      // Todo: It just works
-      queryClient.refetchQueries([`paginated-solutions`, slug]);
+      // invalidate cache on adding a solution successfully
+      queryClient.invalidateQueries({
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-editor.tsx:81]: invalidate on success only?

File: apps/web/src/app/[locale]/challenge/[slug]/solutions/getSolutionRouteData.ts [hunk 3]
```diff
@@ -48,55 +63,13 @@ export const getSolutionsRouteData = cache(async (slug: string, session: Session
...
-export async function getPaginatedSolutions({
-  slug,
-  page,
-  sortKey = 'createdAt',
-  sortOrder = 'desc',
-}: {
-  slug: string;
-  page: number;
-  sortKey?: SortKey;
-  sortOrder?: SortOrder;
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/challenge/[slug]/solutions/getSolutionRouteData.ts:63]: hell yeah

Comment (zaCKoZAck0) [reply apps/web/src/app/[locale]/challenge/[slug]/solutions/getSolutionRouteData.ts:63]: mb

File: apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-editor.tsx (hunk 2 above)
Comment (zaCKoZAck0) [reply apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-editor.tsx:81]: mb

Comment (zaCKoZAck0) [reply apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-editor.tsx:76]: refresh the pr you'll be good to go

**Review by bautistaaa — APPROVED** [review]

**Pull request merged**